File: Modules/module_sample/src/main/java/com/example/william/my/module/sample/contract/ArticleContract.java
package com.example.william.my.module.sample.contract;

import com.example.william.my.library.presenter.IBasePresenter;
import com.example.william.my.library.view.IBaseView;
import com.example.william.my.module.bean.ArticleDetailBean;

import java.util.List;

public interface ArticleContract {

    interface View extends IBaseView<Presenter> {

        void showEmptyView();

        void showArticles(boolean isFirst, List<ArticleDetailBean> article);

        void onDataNoMore();

    }

    interface Presenter extends IBasePresenter {

        void onRefreshArticleList();

        void onLoadMoreArticleList();
    }
}

File: modules/module_sample/src/main/kotlin/com/example/william/my/module/sample/contract/ArticleContract.kt
package com.example.william.my.module.sample.contract

import com.example.william.my.bean.data.ArticleDetailBean
import com.example.william.my.library.presenter.IBasePresenter
import com.example.william.my.library.view.IBaseView

interface ArticleContract {

    interface View : IBaseView<Presenter?> {
        fun showArticles(article: List<ArticleDetailBean?>)
        fun onDataFail()
    }

    interface Presenter : IBasePresenter {
        fun queryArticle(page: Int)
    }
}